src/wallet.js

```diff
@@ -326,9 +349,41 @@ const wallet = {
    * @inner
    */
   isPasswordCorrect(password) {
+    return this.hashValidation(password, 'hashPasswd', 'saltPasswd');
+  },
+
+  /**
+   * Validate if password matches the corresponding hash in the storage
+   *
+   * @param {string} password
+   * @param {string} hashKey key of the hash saved in storage
+   * @param {string} saltKey key of the salt saved in storage
+   *
+   * @return {boolean}
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  hashValidation(password, hashKey, saltKey) {
     const accessData = storage.getItem('wallet:accessData');
```

Comment:
I agree and we've discussed this in the past. The lib needs some refactors but this would affect both desktop and mobile wallet code.

I think we should focus on the new algorithm task and in a different pull request we can implement those refactors in the future.

What do you think?

Reply:
I agree.